python/ray/dag/compiled_dag_node.py

```diff
@@ -841,34 +852,14 @@ def _preprocess(self) -> None:
 
                 # Collect actors for NCCL P2P methods.
                 if dag_node.type_hint.requires_nccl():
-                    nccl_actors_p2p.add(actor_handle)
                     custom_nccl_group = dag_node.type_hint.get_custom_nccl_group()
-                    mixed_nccl_group_error_message = (
-                        "Accelerated DAGs do not support mixed usage of "
-                        "type hints of default NCCL group "
-                        '(i.e., TorchTensor(transport="nccl"))'
-                        "and custom NCCL group "
-                        "(i.e., TorchTensor(transport=nccl_group)). "
-                        "Please check all the TorchTensor type hints and "
-                        "make sure only one type of NCCL transport is specified."
-                    )
-                    if custom_nccl_group is None:
-                        if self._custom_nccl_group_p2p is not None:
-                            raise ValueError(mixed_nccl_group_error_message)
-                        self._use_default_nccl_group = True
+                    if custom_nccl_group is None and self._custom_nccl_group_p2p:
+                        custom_nccl_group = self._custom_nccl_group_p2p
+                    if custom_nccl_group:
+                        custom_nccl_group_to_nodes_p2p[custom_nccl_group].add(dag_node)
                     else:
-                        if self._use_default_nccl_group:
-                            raise ValueError(mixed_nccl_group_error_message)
-                        if self._custom_nccl_group_p2p is not None:
-                            if self._custom_nccl_group_p2p != custom_nccl_group:
-                                raise ValueError(
-                                    "Accelerated DAGs currently only support "
-                                    "a single custom NCCL group, but multiple "
-                                    "have been specified. Check all the "
-                                    "TorchTensor(transport=nccl_group) type hints "
-                                    "to make sure only one NCCL group is used."
-                                )
-                        self._custom_nccl_group_p2p = custom_nccl_group
+                        default_nccl_actors_p2p.add(actor_handle)
+                        default_nccl_nodes_p2p.add(dag_node)
```

Comment:
This code can be cleaner if you keep the original structure that does one pass to collect all of the actors in ops that specified "nccl", then do a separate pass at the end to check that all of these actors are actually in the default group (if specified).

Reply:
Done.